### PR TITLE
MON-137834 compile agent on windows runner

### PR DESCRIPTION
**Pull request by jean-christophe81**
## Description
REFS:MON-137834

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [X] master

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/workflows/centreon-collect.yml [hunk 1]
```diff
@@ -21,10 +21,10 @@ on:
       - cmake.sh
       - cmake-vcpkg.sh
       - CMakeLists.txt
+      - CMakeListsLinux.txt
```
Comment (kduret) [inline .github/workflows/centreon-collect.yml:24]:
```suggestion
```

File: .github/workflows/centreon-collect.yml [hunk 2]
```diff
@@ -21,10 +21,10 @@ on:
...
       - vcpkg.json
       - overlays/**
       - selinux/**
```
Comment (kduret) [inline .github/workflows/centreon-collect.yml:27]:
```suggestion
      - selinux/**
      - vcpkg/**
```

File: .github/workflows/windows-agent.yml [hunk 3]
```diff
@@ -6,9 +6,24 @@ concurrency:
...
+      - CMakeListsWindows.txt
+      - vcpkg.json
+  push:
+    branches:
+      - develop
+      - dev-[2-9][0-9].[0-9][0-9].x
+      - master
+      - "[2-9][0-9].[0-9][0-9].x"
+    paths:
+      - agent/**
```
Comment (kduret) [inline .github/workflows/windows-agent.yml:23]:
```suggestion
      - agent/**
      - custom-triplets/**
      - CMakeLists.txt
      - CMakeListsWindows.txt
      - vcpkg.json
```

File: .github/scripts/windows-agent-compile.ps1 [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+$files_to_hash= "vcpkg.json", "custom-triplets\x64-windows.cmake", "CMakeLists.txt", "CMakeListsWindows.txt"
+$files_content= Get-Content -Path $files_to_hash -Raw
+$stringAsStream = [System.IO.MemoryStream]::new()
+$writer = [System.IO.StreamWriter]::new($stringAsStream)
+$writer.write($files_content -join " ")
+$writer.Flush()
+$stringAsStream.Position = 0
+$vcpkg_hash=Get-FileHash -InputStream $stringAsStream -Algorithm SHA256 | Select-Object Hash
+$file_name = "windows-agent-vcpkg-dependencies-cache-" + $vcpkg_hash.Hash
+$file_name_extension = "${file_name}.7z"
```
Comment (bouda1) [inline .github/scripts/windows-agent-compile.ps1:37]: Be careful to the coding style.
hash= toto
hash = titi
hash=tata

I'd prefer the second one.

File: .github/workflows/centreon-collect.yml (hunk 1 above)
Comment (bouda1) [reply .github/workflows/centreon-collect.yml:24]: Why removing CMakeListsLinux here ?

I think I would also add CMakeListsWindows.txt, but maybe it's due to a misunderstanding on my part.

File: CMakeListsWindows.txt [hunk 5]
```diff
@@ -16,6 +16,11 @@
 # For more information : contact@centreon.com
 #
 
+#When we build from cache (CI), we don't use vcpkg cmketool, so we tell to cmake where to find packages info
```
Comment (bouda1) [inline CMakeListsWindows.txt:19]:
```suggestion
# When we build from cache (CI), we don't use vcpkg cmaketool, so we tell to cmake where to find packages info
```

File: common/grpc/CMakeLists.txt [hunk 6]
```diff
@@ -33,5 +33,6 @@ target_include_directories(centreon_grpc PRIVATE ${INC_DIR})
 
 target_precompile_headers(centreon_grpc REUSE_FROM centreon_common)
 
-set_target_properties(centreon_grpc PROPERTIES COMPILE_FLAGS "-fPIC")
-
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+  set_target_properties(centreon_grpc PROPERTIES COMPILE_FLAGS "-fPIC")
```
Comment (bouda1) [inline common/grpc/CMakeLists.txt:37]: set_target_properties(centreon_grpc PROPERTIES POSITION_INDEPENDENT_CODE ON)

Comment (bouda1) [inline common/grpc/CMakeLists.txt:37]:
```suggestion
  set_target_properties(centreon_grpc PROPERTIES POSITION_INDEPENDENT_CODE ON)
```

Comment (bouda1) [reply common/grpc/CMakeLists.txt:37]: It should work even on windows

File: .github/workflows/centreon-collect.yml (hunk 1 above)
Comment (kduret) [reply .github/workflows/centreon-collect.yml:24]: Indeed it should be kept, I didn't see this file exist
But is should be added to `push.pathes` too

**Review by kduret — APPROVED** [review]

**Review by bouda1 — APPROVED** [review]

**Pull request merged**